### DISCUSSION TITLE
[CategoryAxis] Unknown category issue

### DISCUSSION THREAD

**Discussion by alwgnr** [GSI-CS-CO/chart-fx#435] [Q&A] [answered]
**Bug:**
In my application that is based on the Chart-fx library, which is realy great, the "unknown category" is always shown on the Axis, even though I do not have any anknown category in my data set. 

As a workaround: Is it possible to hide the "unknown category" on the shown axis?

![unknownCategory2](https://user-images.githubusercontent.com/2658682/122953464-0863d600-d37f-11eb-8edc-c3a8485f7515.png)

**To Reproduce**
The same problem I can see if zooming in/out in CategoryAxisSample:
https://github.com/GSI-CS-CO/chart-fx/blob/master/chartfx-samples/src/main/java/de/gsi/chart/samples/CategoryAxisSample.java

![unknownCategory](https://user-images.githubusercontent.com/2658682/122948048-e5cfbe00-d37a-11eb-8c3a-d81338af2c0d.png)

 
**Environment:**
 - OS: Windows 10
 - Java version: 11
 - JavaFx version: 16
 - ChartFx version: 11.2.5

**Comment by RalphSteinhagen** [comment, marked as answer]
@alzorin this is very similar to your second question [here](https://github.com/GSI-CS-CO/chart-fx/discussions/434).

The 'unknown category' is the designed behaviour and default label defined in the `CategoryAxis`
https://github.com/GSI-CS-CO/chart-fx/blob/f72b16aa85f7c50ed17aab3b5c1650a2ffbbcca9/chartfx-chart/src/main/java/de/gsi/chart/axes/spi/CategoryAxis.java#L145-L152
which is triggered if you zoom or set a range outside the defined categories. The rationale behind this is to clearly mark the valid range and to provide an overflow bin for values that do not fit any other category.

You may change this via limiting the range to something that is just smaller/larger than your defined range, provide extra user-defined (e.g. "") categories (works only for right/top range), or overwrite the above method with a user-level derived class as also outlined [here](#434).

<sub>
P.S. May I ask for which application/domain are you using chart-fx? Is it a private/academic/commercial project? Am asking because it's hard to get a good overview of the different use-cases and user-base via GitHub.
</sub>

**Reply by alwgnr** [reply]
@RalphSteinhagen Thank you for the fast and full answer. I use chart-fx for the field of process mining it is a academic project. ChartFx solves great the performance bottleneck of the standard JavaFx chart library and adds important features like Zoomer and Panner. This is the main reason why I found the ChartFx library.
